src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -1965,6 +1968,27 @@ private bool LowerBoundInterfaceInference(TypeSymbol source, NamedTypeSymbol tar
             return true;
         }
 
+        internal static ImmutableArray<NamedTypeSymbol> ModuloNullabilityDifferences(IEnumerable<NamedTypeSymbol> interfaces, VarianceKind variance)
+        {
+            var dictionary = new Dictionary<NamedTypeSymbol, NamedTypeSymbol>(SymbolEqualityComparer.IgnoreNullableModifiersForReferenceTypes);
```

Comment:
Consider using `PooledDictionary`

Reply:
Or consider using a HashSet

---
In reply to: [321005507](https://github.com/dotnet/roslyn/pull/38460#discussion_r321005507) [](ancestors = 321005507)

Reply:
The default  pool doesn't have the right comparer. I considered creating a separate pool then backtracked. I'll put it back in.